client/src/components/Navbar/Navbar.js

```diff
@@ -1,21 +1,114 @@
-import React from 'react';
+import React, { Component } from "react";
 import "./Navbar.css";
-import Logo from "../Logo/Logo"
+import Logo from "../Logo/Logo";
+import Burger from "../Burger/Burger";
+import ProfilePicture from "../ProfilePicture/ProfilePicture";
+import NavLink from "../NavLink/NavLink";
+import Icon from "../Icon/Icon";
+import PropTypes from "prop-types";
 
-// Function based React Component
-const Navbar = (props) => {
+// Class Based React Component
+class Navbar extends Component {
+  constructor(props) {
+    super(props);
+    // Default CSS class to apply to the Component
+    this.state = {
+      condensedMenuActive: false,
+      condensedMenu: "",
+      // pageMark controls the triangle that marks which page the user is on.
+      // Passing 0-3 as a prop into this component controls where it displays.
+      pageMark: ["", "", "", ""],
+      // menuItems are careated in order of the following array.
+      menuItems: ["My Directory", "Community", "Forum", "Book a Trip"],
+      // The link end points for each menu item.
+      menuLinks: ["/", "/", "/", "/"],
+      // Each item in this array is a set for the dropdown on each menu link.
+      // This is an example for one drop menu item: { text: 'Explore People', to: '/' }
+      menuDropDown: [[], [], [], []]
+    };
+  }
 
-  // Default Class to apply to Component
-  let classList = `Navbar`;
+  // Handles activating the hamburger animation and displays the menu.
+  handleCondensedMenuClick = () => {
+    this.setState({ condensedMenuActive: !this.state.condensedMenuActive });
+    if (this.state.condensedMenuActive === false) {
+      this.setState({ condensedMenu: "Navbar__show" });
+    } else {
+      this.setState({ condensedMenu: "" });
+    }
+  }
 
+  // Sets the location of the triangle on the menu.
+  setPageMarker = () => {
+    let setPage = [];
+    for (let i = 0; i < 4; i++) {
+      if (i === this.props.page) {
+        setPage.push("Navbar__show");
+      } else {
+        setPage.push("");
+      }
+    }
+    this.setState({ pageMark: setPage });
```

Comment:
This should also be done in render

Reply:
I also get the ```Maximum update depth exceeded.``` error when I do this.

Reply:
Due to how I had to handle some things with the triangle, I really do feel that I need this function. It's now needed for the click handler on the burger icon to help manage displaying the triangle.

Reply:
Could the nav bar component accept a prop (i.e. from React Router) for the current path and then use that to determine which nav link is "active"? That would make this component much simpler

Reply:
@ZakkMan I have tried putting this in several places in render and the triangle just doesn't display when I do it that way. I was able to get everything else to be in the render section, though.

Reply:
> Could the nav bar component accept a prop (i.e. from React Router) for the current path and then use that to determine which nav link is "active"? That would make this component much simpler

I missed this note. I'll look into that.

Reply:
@ZakkMan I'm only seeing how to do this if <Route> is calling to render that particular component and using the component attribute.  Route isn't creating the Navbar, right? I'm not sure if that's what you're getting at, though. Right now I have a prop being passed into Navbar that tells it which page is active. I'm using that to put a class in in one spot of an array which gets iterated through to one of the navlinks. I can also make the each array element blank if I want to triangle not to display in the case of the burger menu. I'm not entirely sure how to simplify it. I've been researching it a lot but I can keep looking.

Reply:
This is the only one that I can't figure out how to get rid of at this point. I have now condensed everything else.